### PR TITLE
add KeepCRV tracking.

### LINKED ISSUE #20: Add support to detect keepCRV field
Some contract have keepCRV setting that we want to track and monitor, this is another field that is not generic so we need a similar approach to https://github.com/yearn/yearn-watch-legacy/issues/74 for this field.

Ideally the implementation can pave the road to making it easy to add new fields as we need to monitor, in this case it may be solved by checking a regex or CRV string in strategy name to detect that is most likely to have the field

https://github.com/yearn/yearn-watch-legacy/issues/75
https://github.com/yearn/yearn-watch-legacy/pull/331

### PR DESCRIPTION
**Pull request by CutupAngel**

### CODE REVIEW

File: components/sections/strategies/SectionStats.tsx [hunk 1]
```diff
@@ -39,6 +39,7 @@ const	SectionStats = React.memo(function SectionStats({currentVault, currentStra
 					{title: 'Performance Fee ', details: format.bigNumberAsAmount(currentStrategy.performanceFee, 2, 2, '%')},
 					{title: 'Min Debt Per Harvest ', details: format.bigNumberAsAmount(currentStrategy.minDebtPerHarvest, currentVault.decimals, 4)},
 					{title: 'Max Debt Per Harvest ', details: format.bigNumberAsAmount(currentStrategy.maxDebtPerHarvest, currentVault.decimals, 4)},
+					{title: 'KeepCRV ', details: format.bigNumberAsAmount(currentStrategy.keepCRV, 0)},
```
Comment (magic990619) [inline components/sections/strategies/SectionStats.tsx:42]: If the strategy doesn't have `keepCRV` field, it should display `N/A`.

image

File: contexts/useWatch.d.tsx [hunk 2]
```diff
@@ -89,6 +89,7 @@ export type TStrategy = {
 	expectedReturn: BigNumber,
 	reports: TStrategyReport[],
 	alerts: TAlert[],
+	keepCRV: BigNumber,
```
Comment (magic990619) [inline contexts/useWatch.d.tsx:92]: It should be `string` since it is in need to show `N/A` for some strategies.

File: pages/api/getVaults.tsx [hunk 3]
```diff
@@ -395,6 +396,8 @@ export async function getVaults(
 			strategy.expectedReturn = utils.format.BN(callResult?.[rIndex++] as never);
 			strategy.isActive = callResult[rIndex++] as boolean;
 			strategy.estimatedTotalAssets = utils.format.BN(callResult?.[rIndex++] as never);
+			strategy.keepCRV = utils.format.BN(callResult?.[rIndex++] as never);
```
Comment (magic990619) [inline pages/api/getVaults.tsx:399]: Need to format to `string`

File: pages/track.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+			for (const strategy of vault.strategies) {
+				const	keepCRV = strategy.keepCRV;
+
+				if (!keepCRV)
+					continue;
+				_filteredStrategies.push(strategy);
+			}
+		}
+		set_filteredStrategies(_filteredStrategies);
+	}, [vaults, searchTerm, isOnlyWithTvl]);
```
Comment (magic990619) [inline pages/track.tsx:72]: `isOnlyWithTVL` is not implemented.

File: components/sections/strategies/SectionStats.tsx (hunk 1 above)
Comment (magic990619) [reply components/sections/strategies/SectionStats.tsx:42]: This kinda strategies should not be tracked on `Track` page.

Comment (CutupAngel) [reply components/sections/strategies/SectionStats.tsx:42]: Then, what page is good for tracking keepcrv?
I created new page named Track by myself.

I'd like to know your opinion.

Comment (magic990619) [reply components/sections/strategies/SectionStats.tsx:42]: `Track` page is the correct one where we should track strategies with `keepCRV` field.
What I am saying is that, currently it has some strategies without `keepCRV` field.
Like I mentioned in the above image.

Comment (CutupAngel) [reply components/sections/strategies/SectionStats.tsx:42]: Is this none `keepCRV` strategy from track page?

Comment (magic990619) [reply components/sections/strategies/SectionStats.tsx:42]: [Lido Eth 2.0 Staking](https://etherscan.io/address/0x0967aFe627C732d152e3dFCAdd6f9DBfecDE18c3#readContract) doesn't have `keepCRV` field, but it is on `Track` page.

image

**Review by magikadev — APPROVED** [review]

**Pull request merged**